extensions/advanced_ballistics/AdvancedBallistics.cpp

```diff
@@ -114,117 +115,56 @@ double calculateAtmosphericCorrection(double ballisticCoefficient, double temper
     }
 }
 
-double calculateRetard(int DragFunction, double DragCoefficient, double Velocity) {
-    double vel = Velocity * 3.2808399;
-    double val = -1;
-    double A = -1;
-    double M = -1;
+double calculateRetard(int DragFunction, double DragCoefficient, double Velocity, double Mach) {
+    std::vector<double> machNumbers = {};
```

Comment:
Performance optimization would be to have these tables be built at compile-time.
Or making them global variables and just selecting a pointer to them here.
Because this way you have a memory allocation and memcpy everytime a drag function is choosen.